### ISSUE TITLE
Kinopoisk  Статус: Malfunctioned

### ISSUE THREAD

**Issue by Akior** [LinFor/jellyfin-plugin-kinopoisk#7]
Не могут загрузиться несколько dll на свежих версиях jelly, плагин не работает

Failed to load assembly "/var/lib/jellyfin/plugins/РљРёРЅРѕРџРѕРёСЃРє_10.7.5.1/Newtonsoft.Json.dll". Disabling plugin.
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
[2021-07-28 17:26:22.970 +03:00] [ERR] Failed to load assembly "/var/lib/jellyfin/plugins/РљРёРЅРѕРџРѕРёСЃРє_10.7.5.1/KinopoiskUnofficialInfo.ApiClient.dll". Disabling plugin.
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (0x80131621)
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
 ---> System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at System.Reflection.Assembly.LoadFromResolveHandler(Object sender, ResolveEventArgs args)
   at System.Runtime.Loader.AssemblyLoadContext.InvokeResolveEvent(ResolveEventHandler eventHandler, RuntimeAssembly assembly, String name)
   at System.Runtime.Loader.AssemblyLoadContext.OnAssemblyResolve(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
[2021-07-28 17:26:22.971 +03:00] [ERR] Failed to load assembly "/var/lib/jellyfin/plugins/РљРёРЅРѕРџРѕРёСЃРє_10.7.5.1/Jellyfin.Plugin.Kinopoisk.dll". Unknown exception was thrown. Disabling plugin.
System.IO.FileNotFoundException: Could not load file or assembly 'MediaBrowser.Common, Version=10.7.6.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'MediaBrowser.Common, Version=10.7.6.0, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()

**Comment by mastercho** [comment]
Yes i got same issue on Jellyfin 10.7.6

**Comment by Akior** [comment]
> Любопытно как. Собрал свежий Jellyfin, а там всё работает.
> 
> Скажите точно, откуда установлен? Докер? Точный тег образа?

Версия: 10.7.6
Установлена через apt из репы 
deb [arch=amd64] https://repo.jellyfin.org/ubuntu focal main

Версия кп:
КиноПоиск10.7.5.2
Статус: Malfunctioned

Сейчас ошибка такая:

Error creating "Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonImageProvider"
System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonImageProvider'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
[2021-08-10 21:18:09.402 +03:00] [ERR] Error creating "Jellyfin.Plugin.Kinopoisk.MetadataProviders.VideoImageProvider"
System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.VideoImageProvider'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
[2021-08-10 21:18:09.430 +03:00] [ERR] Error creating "Jellyfin.Plugin.Kinopoisk.MetadataProviders.MovieMetadataProvider"
System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.MovieMetadataProvider'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
[2021-08-10 21:18:09.431 +03:00] [ERR] Error creating "Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonMetadataProvider"
System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonMetadataProvider'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
[2021-08-10 21:18:09.431 +03:00] [ERR] Error creating "Jellyfin.Plugin.Kinopoisk.MetadataProviders.SeriesMetadataProvider"
System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.SeriesMetadataProvider'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)

**Comment by mastercho** [comment]
> Любопытно как. Собрал свежий Jellyfin, а там всё работает.
> 
> Скажите точно, откуда установлен? Докер? Точный тег образа?

Docker https://hub.docker.com/r/linuxserver/jellyfin/

Версия: 10.7.6

Операционна система: Linux

Архитектура: X64

**Comment by iAHTOH** [comment]
Есть понимание на что смотреть? Не работает.

```
 [13:00:31] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.Kinopoisk, Version=10.7.5.2, Culture=neutral, PublicKeyToken=null from /config/data/plugins/КиноПоиск_10.7.5.2/Jellyfin.Plugin.Kinopoisk.dll
 [13:00:31] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Failed to load assembly /config/data/plugins/КиноПоиск_10.7.5.2/Newtonsoft.Json.dll. Disabling plugin.
 System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
    at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
    at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
    at System.Reflection.Assembly.LoadFrom(String assemblyFile)
    at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
 [13:00:41] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: OMDb 10.7.7.0
 [13:00:41] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: MusicBrainz 10.7.7.0
 [13:00:41] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: AudioDB 10.7.7.0
 [13:00:41] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonImageProvider
 System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.PersonImageProvider'.
    at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
    at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
    at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
 [13:00:41] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Jellyfin.Plugin.Kinopoisk.MetadataProviders.VideoImageProvider
 System.InvalidOperationException: Unable to resolve service for type 'KinopoiskUnofficialInfo.ApiClient.IKinopoiskApiClient' while attempting to activate 'Jellyfin.Plugin.Kinopoisk.MetadataProviders.VideoImageProvider'.
    at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
    at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
    at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type)
```

**Comment by freefd** [comment]
Hi, faced the same issue with КиноПоиск 10.7.5.2 plugin after upgrading Jellyfin container instance from 10.7.2 -1 to 10.7.7-1. The build is from https://hub.docker.com/r/linuxserver/jellyfin folks.

I even tried to reinstall КиноПоиск plugin but it doesn't help at all. Moreover, Jellyfin throws an error like it cannot find `Netonsoft.Json.dll` file:
```
[01:54:43] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Failed to load assembly /config/data/plugins/КиноПоиск_10.7.5.2/Newtonsoft.Json.dll. Disabling plugin.
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
[01:54:43] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Jellyfin.Plugin.Kinopoisk, Version=10.7.5.2, Culture=neutral, PublicKeyToken=null from /config/data/plugins/КиноПоиск_10.7.5.2/Jellyfin.Plugin.Kinopoisk.dll
[01:54:43] [ERR] [1] Emby.Server.Implementations.Plugins.PluginManager: Failed to load assembly /config/data/plugins/КиноПоиск_10.7.5.2/KinopoiskUnofficialInfo.ApiClient.dll. Disabling plugin.
System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. Could not find or load a specific file. (0x80131621)
File name: 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
 ---> System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromPath(IntPtr ptrNativeAssemblyLoadContext, String ilPath, String niPath, ObjectHandleOnStack retAssembly)
   at System.Runtime.Loader.AssemblyLoadContext.LoadFromAssemblyPath(String assemblyPath)
   at System.Reflection.Assembly.LoadFrom(String assemblyFile)
   at System.Reflection.Assembly.LoadFromResolveHandler(Object sender, ResolveEventArgs args)
   at System.Runtime.Loader.AssemblyLoadContext.InvokeResolveEvent(ResolveEventHandler eventHandler, RuntimeAssembly assembly, String name)
   at System.Runtime.Loader.AssemblyLoadContext.OnAssemblyResolve(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()
```
but this file is definitely there:
```
# docker exec -ti jellyfin bash -c 'ls -la /config/data/plugins/КиноПоиск_10.7.5.2/'
total 896
drwxr-xr-x 2 abc abc   4096 Sep 24 01:54 .
drwxr-xr-x 4 abc abc   4096 Sep 24 01:54 ..
-rw-r--r-- 1 abc abc   5067 Sep 24 01:54 25f8c0315e6ccb2aa6c2642e48f2c9e9.png
-rw-r--r-- 1 abc abc  51712 Sep 24 01:54 Jellyfin.Plugin.Kinopoisk.dll
-rw-r--r-- 1 abc abc 146944 Sep 24 01:54 KinopoiskUnofficialInfo.ApiClient.dll
-rw-r--r-- 1 abc abc   2290 Sep 24 01:54 meta.json
-rw-r--r-- 1 abc abc 695336 Sep 24 01:54 Newtonsoft.Json.dll
```

Any suggestions, fixes please? Thank you.

**Comment by mastercho** [comment]
I think this plugin is deprecated as dev wont work on it anymore

**Comment by freefd** [comment]
@mastercho please do not rush, let's hear out the author first :)

**Comment by LinFor** [comment]
@Akior @mastercho @freefd Please try new 10.7.5.3 version. I have removed a separate version of Newtonsoft.Json and hopefully will pick up the one in Jellyfin.

**Comment by Akior** [comment]
> @Akior @mastercho @freefd Please try new 10.7.5.3 version. I have removed a separate version of Newtonsoft.Json and hopefully will pick up the one in Jellyfin.
Не работает 

[2021-09-27 15:36:11.666 +03:00] [ERR] Failed to load assembly "/var/lib/jellyfin/plugins/РљРёРЅРѕРџРѕРёСЃРє_10.7.5.3/Jellyfin.Plugin.Kinopoisk.dll". Unknown exception was thrown. Disabling plugin.
System.IO.FileNotFoundException: Could not load file or assembly 'MediaBrowser.Common, Version=10.7.7.0, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.

File name: 'MediaBrowser.Common, Version=10.7.7.0, Culture=neutral, PublicKeyToken=null'
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at Emby.Server.Implementations.Plugins.PluginManager.LoadAssemblies()+MoveNext()

**Comment by freefd** [comment]
@LinFor thanks a lot for the new version provided. I've just tested it on the local Jellyfin ephemeral container and now it seems to be working fine from integration pov:
![image](https://user-images.githubusercontent.com/133116/134909132-71c02c92-d8a2-4fef-aadc-2f1144ce8dde.png)

But I cannot test with a real data collecting from Кинопоиск due to my original installation isn't available to me until this Friday (I hope).

**Comment by svyaznoy362** [comment]
> @Akior @mastercho @freefd Попробуйте новую версию 10.7.5.3. Я удалил отдельную версию Newtonsoft.Json и, надеюсь, сохраню версию в Jellyfin.

Добрый день. Не работаю трейлеры в клиенте.

**Comment by LinFor** [comment]
@svyaznoy362 киньте пожалуйста id или ссылку на кинопоиск с проблемным фильмом. И, лучше в отдельный issue.

**Comment by mastercho** [comment]
Fixed and working on Jellyfin 10.7.7 Спосиба!
